src/App.tsx

```diff
@@ -1,26 +1,78 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import { BATCH_DUMP_TIMEOUT_MS, FilesApi } from './api-client';
+
+const fileIds = ['fileid1', 'fileid2', 'fileid3'];
+
+const App = () => {
+  let combinedFiles: string[] | undefined;
```

Comment:
I am not proud of this. Because requestFile gets promise response at the same time you can't easily combine files payload. I decided to go with this. So I have this variable outside of the closure and use it as an intermediary to combine files. On every render it's reset.

In a real life it's possible that we would have had a context and that would have been handled there without this dirt. I thought about context but thought that for such a small app it's an overkill.